core/forms/MemberForms.py

```diff
@@ -58,6 +99,33 @@ def clean_password2(self):
             raise forms.ValidationError("Passwords don't match")
         return password2
 
+    def clean_membership(self):
+        """
+        Convert the membership selection into a timedelta, and verify it is valid for the members current status
+        """
+        selection = self.cleaned_data['membership']
+
+        # Convert the membership selection into a timedelta
+        if selection == "60" or selection == "40":
```

Comment:
I think it's better to cast the cleaned data to an int than converting 60 and 40 to strings

Reply:
And also what do these numbers mean? I remember now, but future me won't. Extract them to consts

Reply:
Good points. Fixed both by using strings for the various membership choices (year_new, quarter_return etc) , also allowed me to clean up some of the logic